posts/2021-08-30-binding-app-to-postgresql-db-with-odo-and-ol-stack.adoc

```diff
@@ -0,0 +1,284 @@
+
+
+=== Install the needed operators
+
+If you have the right privileges, you can use the OpenShift console to install these operators. You do this by navigating to `Operators->OperatorHub` and searching/selecting/installing a couple of operators from the catalog.
+
+- Service Binding Operator (provided by Red Hat)
+
+Use the OpenShift console (`Operators->OperatorHub`) to install this operator. Be sure to install version 0.9.1 and above.
+
+- PostgreSQL Operator (provided by Dev4Ddevs.com)
+
+Before installing this operator, create the project/namespace under which the application will be deployed. This operator is namespace scoped, so it needs to be installed in the same namespace where the application will be deployed.
+
+[source,sh]
+----
+odo project create service-binding-demo
+----
+
+Use the OpenShift console (`Operators->OperatorHub`) to install this operator. Be sure to pick `service-binding-demo` as the namespace in which to install it.
+
+=== Create a Java Open Liberty based component and the Database resource service
+
+Clone the application repository.
+
+[source,sh]
+----
+git clone https://github.com/OpenLiberty/application-stack-samples.git && \
+cd application-stack-samples/jpa
+----
+
+Create a Java Open Liberty component.
+
+[source,sh]
+----
+odo create java-openliberty mysboproj
+----
+
+Display the service providers and services available in the cluster.
+
+[source,sh]
+----
+odo catalog list services
+----
+
+Output:
+
+[source,sh]
+----
+Services available through Operators
+NAME                                CRDs
+...
+postgresql-operator.v0.1.1          Backup, Database
+...
+----
+
+Generate the `postgresql-operator.v0.1.1` provided Database resource/service yaml and store it in a file. The data is stored in a file because some entries such as the resource name, the access information, and annotations will be customized.
+
+[source,sh]
+----
+odo service create postgresql-operator.v0.1.1/Database --dry-run > db.yaml
+----
+
+Customize the database name, user name, and password values under the spec section in `db.yaml`.
+
+[source,sh]
+----
+  databaseName: "sampledb"
+  databasePassword: "samplepwd"
+  databaseUser: "sampleuser"
+----
+
+Customize the Database resource name and add the needed annotations in `db.yaml`. The annotations are used to inform the Service Binding Operator about the service data required by the application.
+
+[source,sh]
+----
+metadata:
+  name: sampledatabase
+  annotations:
+    service.binding/db_name: 'path={.spec.databaseName}'
+    service.binding/db_password: 'path={.spec.databasePassword}'
+    service.binding/db_user: 'path={.spec.databaseUser}'
+----
+
+Generate the Database resource service devfile configuration.
+
+[source,sh]
+----
+odo service create --from-file db.yaml
+----
+
+Configuration added to devfile.yaml:
```

Comment:
Configuration added to devfile.yaml:
--->
The following configuration is automatically added to `devfile.yaml`: